tests/backup/backup_upgrade_test.go

```diff
@@ -204,3 +206,419 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
+			postRuleUids = make(map[string]string)
+			log.InfoD("Creating post-exec rules for applications %v", Inst().AppList)
+			for _, appName := range Inst().AppList {
+				log.Infof("Creating post backup rule for application [%s]", appName)
+				_, postRuleName, err := Inst().Backup.CreateRuleForBackup(appName, orgID, "post")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of post-backup rule for application [%s]", appName))
+				postRuleUid := ""
+				if postRuleName != "" {
+					postRuleUid, err = Inst().Backup.GetRuleUid(orgID, ctx, postRuleName)
+					log.FailOnError(err, "Fetching post backup rule [%s] uid", postRuleName)
+					log.Infof("Post backup rule [%s] uid: [%s]", postRuleName, postRuleUid)
+				}
+				for i := 0; i < len(srcClusterAppNamespaces[appName]); i++ {
+					postRuleNames[appName] = postRuleName
+					postRuleUids[appName] = postRuleUid
+				}
+			}
+		})
+		Step("Create backups with and without pre- and post-exec rules", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			createBackupWithRulesTask := func(appName string) {
+				namespace := srcClusterAppNamespaces[appName][0]
+				backupName := fmt.Sprintf("%s-%s-%v-with-rules", BackupNamePrefix, namespace, time.Now().Unix())
+				labelSelectors := make(map[string]string, 0)
+				log.InfoD("Creating a backup of namespace [%s] with pre- and post-exec rules", namespace)
+				err := CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUid, []string{namespace},
+					labelSelectors, orgID, srcClusterUid, preRuleNames[appName], preRuleUids[appName], postRuleNames[appName], postRuleUids[appName], ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup [%s]", backupName))
+				backupWithRuleNames = SafeAppend(&mutex, backupWithRuleNames, backupName).([]string)
+			}
+			_ = TaskHandler(Inst().AppList, createBackupWithRulesTask, Parallel)
+			createBackupWithoutRulesTask := func(appName string) {
+				namespace := srcClusterAppNamespaces[appName][0]
+				backupName := fmt.Sprintf("%s-%s-%v-without-rules", BackupNamePrefix, namespace, time.Now().Unix())
+				labelSelectors := make(map[string]string, 0)
+				log.InfoD("Creating a backup of namespace [%s] without pre- and post-exec rules", namespace)
+				err := CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUid, []string{namespace},
+					labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup [%s]", backupName))
+				backupWithoutRuleNames = SafeAppend(&mutex, backupWithoutRuleNames, backupName).([]string)
+			}
+			_ = TaskHandler(Inst().AppList, createBackupWithoutRulesTask, Parallel)
+		})
+		Step("Create a schedule policy", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			intervalInMins = 5
```

Comment:
I will set intervalInMins to 15 at the end of review.

Reply:
As we are not testing UI, better keep this interval as 5 mins. However try to take many schedules before upgrade.

Reply:
Make sense. Just to be safe I set it back to 15 mins.